### DISCUSSION TITLE
Vercel deployment type compilation not matching local? "does not exist on type 'never'"

### DISCUSSION THREAD

**Discussion by ThomasGormley** [vercel/community#825] [Help] [answered]
Hi,

Typescript compilation and `next build` locally is fine, however in Vercel deployment environment TypeScript is having a hard time with what seems to be nested predicates?

e.g. using a predicate function, it cannot identify the negated logical step

```ts
const isType = isObjType(obj)
//jsx
return (
<p>{isType ? obj.hasProp : obj.otherTypeProp }</p>
);

```

Will fail Vercel deployment with 
`Type error: Property 'otherTypeProp ' does not exist on type 'never'.`

Also noticed that nested logical checks weren't correctly being checked in Vercel env with no issues e.g.

```ts
function getNumberOfRepliesFromCommentType(post: Comment | Listing): number {
    const isListing = isListingType(post);
    if (isListing) {
        const newPost = post.data?.children[0];
        const isComment = isCommentType(newPost);
        if (isComment) {
            return getNumberOfRepliesFromCommentType(newPost);
        } else {
            throw Error("Why is a t3_ type Link here?");
        }
    }
    // hasReplies can only be of type comment as we either Error, or recursively retry until it is a comment,
   //  but Vercel fails with Type error: Property 'data' does not exist on type 'never'.
    const hasReplies = Boolean(post.data.replies);
// ...
```

edit: Just tried using brackets notation instead of dot and it "fixed" the issue - I can deal with that however locally this issue doesn't show, which is an issue

Thanks

**Comment by amyegan** [comment, marked as answer]
I'm glad you found a way to make it work, but it is strange that you were getting different behavior on local and Vercel. If you can share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I would love to take a closer look and figure out what is causing it.

**Reply by ThomasGormley** [reply]
I managed to resolve the issue entirely without having to scrap dot notation everywhere in the code base. I didn't do much exploration into the cause but it seems like it stemmed from a commit where I had replaced all my imports to `tsconfig` aliases.

I'll try reproduce it again in its own branch.
